lib/httpHandler.ts

```diff
@@ -0,0 +1,53 @@
+import { EventWithMiddlewareContext } from "somod";
+import {
+  Copy,
+  CustomEventType,
+  HttpResponse,
+  LambdaFunctionType
+} from "./types";
+
+export class HttpLambda {
+  private apis: Record<string, LambdaFunctionType> = {};
+
+  register = <TBody = Record<string, unknown>>(
+    routeKey: string,
+    lambdaFn: LambdaFunctionType<TBody>
+  ) => {
+    /**
+     * as LambdaFunctionType is just hack to get types mapped properly
+     */
+    this.apis[routeKey] = lambdaFn as LambdaFunctionType;
+  };
+
+  getHandler = () => {
+    return async (
+      event: EventWithMiddlewareContext<Copy<CustomEventType>>
+    ): Promise<HttpResponse> => {
+      const lambdaFn = this.apis[event.routeKey.toLocaleLowerCase()];
+      if (lambdaFn == undefined) {
+        throw new Error("404, url dose not exists");
```

Comment:
instead of throw, it should return a response as per aws payload format version 2

Reply:
done